Sass Refactor: Migration from @import to @use

c4rlosviteri opened the pull request:
# Sass Refactor: Migration from @import to @use

- Migrated all relevant Sass files from the legacy @import syntax to the modern @use syntax, following best practices for modular and maintainable stylesheets.
- Carefully reviewed and updated global and module styles to avoid double-importing files, ensuring no style duplication or specificity conflicts.
- Refactored imports to leverage Sass modules, improving encapsulation and reducing the risk of variable/function collisions.
- Verified that all style dependencies are loaded only once per build, optimizing CSS output and preventing redundant code.
- Maintained style specificity and visual consistency across all components, with special attention to global vs. module scope.
- Updated build process to support the new @use -based Sass architecture.
Testing & Verification

- Built and tested the new styles version to ensure all components render correctly and no regressions were introduced.
- Confirmed that users can now use @use instead of @import for all style modules.

---

This PR modernizes the project's Sass architecture, paving the way for better scalability and maintainability of styles.

mhewson commented:
approved

cviterirei committed:
refactor(styles): migrate SCSS imports to @use and update tokens dependency

- Replace @import with @use for better module encapsulation
- Update @rei/cdr-tokens dependency from ^12.7.0 to ^12.7.1
- Remove unused imports and clean up SCSS files
- Standardize token variable references using tokens namespace
- Delete deprecated CdrPresets.module.scss file

mhewson commented on src/components/abstract/styles/CdrAbstract.module.scss:
```diff
@@ -1,17 +1,16 @@
-@import '../../../styles/settings/index';
-@import '../../../styles/settings/fluid.vars';
+@use '/node_modules/@rei/cdr-tokens/dist/rei-dot-com/scss/cdr-tokens' as tokens;
```
I think we can remove the "/node_modules/" from the @use path --  `@use '@rei/cdr-tokens/dist/rei-dot-com/scss/cdr-tokens'` as tokens;

mhewson replied:
I also wonder if it would be useful to create a css partial that we could use to centralize this path and then do something like 

`@use 'src/styles/tokens' `as tokens;
in the actual files

mhewson commented on src/components/abstract/styles/CdrAbstract.module.scss:
```diff
@@ -1,17 +1,16 @@
 
 .cdr-abstract {
-  @include cdr-text-subheading-sans-500;
+  @include tokens.cdr-text-subheading-sans-500;
```
I didn't call this out above but lets change the as tokens to as cdrTokens

`@include cdrTokens.cdr-text-subheading-sans-500;`

mhewson commented on src/styles/settings/space.vars.scss:
```diff
@@ -1,20 +1,22 @@
 // avoid possible confusion for users, and limit classes generated.
+@use '/node_modules/@rei/cdr-tokens/dist/rei-dot-com/scss/cdr-tokens' as tokens;
+
 $spacing: (
```
I'll say this every time I see this - but these need to come from the tokens rather than be hardcoded in cedar

c4rlosviteri replied:
But at this point we don't have tokens separated by category in cedar-tokens

mhewson replied:
correct- but I think it should be prioritized

c4rlosviteri replied:
tracked in https://rei-coop.atlassian.net/browse/CDR-3273

mhewson commented on src/styles/settings/border.vars.scss:
```diff
@@ -1,11 +1,13 @@
+@use '/node_modules/@rei/cdr-tokens/dist/rei-dot-com/scss/cdr-tokens' as tokens;
```
were these accessing tokens from main before?

c4rlosviteri replied:
From settings/index

mhewson commented on src/styles/settings/index.scss:
```diff
@@ -2,14 +2,13 @@ $cdr-warn: false;
 
 @use 'sass:map';
 @use 'sass:list';
-@import '../../../node_modules/@rei/cdr-tokens/dist/rei-dot-com/scss/cdr-tokens';
```
why remove this and then reference it in each of the included files

c4rlosviteri replied:
Because this import creates a copy of the content of the file in this one, we dont need that.

mhewson commented on src/styles/main.scss:
```diff
@@ -1,6 +1,3 @@
-/* Tokens */
-@import '/node_modules/@rei/cdr-tokens/dist/rei-dot-com/scss/cdr-tokens';
```
lets verify that main.scss is not used by microsite teams to access tokens locally on their projects

mhewson commented on src/components/tooltip/styles/CdrTooltip.module.scss:
```diff
@@ -1,7 +1,4 @@
-@import '../../../styles/settings/index';
-
-//@import url('@rei/cedar/dist/style/cdr-popup.css');
-@import '../../popup/styles/CdrPopup.module';
```
How are the Popup styles accessible to tooltip?

c4rlosviteri replied:
Through the Tooltip component

mhewson commented on src/components/banner/styles/CdrBanner.module.scss:
```diff
@@ -1,8 +1,9 @@
-@import './vars/CdrBanner.vars';
+@use '/node_modules/@rei/cdr-tokens/dist/rei-dot-com/scss/cdr-tokens' as tokens;
+
+@use './vars/CdrBanner.vars' as vars;
```
these vars files should be renamed mixins and used as mixin

mhewson commented on src/components/banner/styles/CdrBanner.module.scss:
```diff
@@ -1,8 +1,9 @@
 
 .cdr-banner {
-  @include cdr-banner-base-mixin;
+  @include vars.cdr-banner-base-mixin;
```
eventually 
`@include mixin.cdr-banner-base;`

mhewson commented on src/components/accordion/styles/CdrAccordion.module.scss:
```diff
@@ -1,38 +1,32 @@
-@import '../../../styles/settings/index';
-
-//@import url('@rei/cedar/dist/style/cdr-icon.css');
-@import '../../icon/styles/CdrIcon.module';
```
how is accordion accessing icons now? 
in the Cedar repo our components can access these as we load all cedar styles but on a project they would only add accordion and not know that the icon and accordion group were also needed

c4rlosviteri replied:
In the icon component.

mhewson commented on src/components/accordion/styles/CdrAccordion.module.scss:
```diff
@@ -1,38 +1,32 @@
-@import '../../icon/styles/CdrIcon.module';
-
-//@import url('@rei/cedar/dist/style/cdr-accordion-group.css');
-@import './CdrAccordionGroup.module';
```
where is this being imported now?

c4rlosviteri replied:
Not necessary, it's already in the accordion group component and icon component.

cviterirei committed:
refactor(styles): replace @import with @use for SCSS modules

Update SCSS files to use @use instead of @import for better module system support
Standardize cdr-tokens import path across all files by removing node_modules prefix

cviterirei committed:
Add style dependencies back

c4rlosviteri committed:
Merge branch 'main' into pr/sass-use

Commit:
Move @use statements to to the top per linting rules

Commit:
Bump version to 15.6.1-beta.5
